src/main/java/slimeknights/tconstruct/tools/tools/Scythe.java

```diff
@@ -1,158 +1,306 @@
+    if(player.getCooledAttackStrength(0.5F) <= 0.9f) {
+      return super.onLeftClickEntity(stack, player, target);
+    }
+
+    // increase the size based on the AOE stuffs
+    TinkerToolEvent.ExtraBlockBreak event = TinkerToolEvent.ExtraBlockBreak.fireEvent(stack, player, player.worldObj.getBlockState(target.getPosition()), 3, 3, 3, -1);
+    if(event.isCanceled()) {
+      return false;
+    }
+
     // AOE attack!
+    player.worldObj.playSound(null, player.posX, player.posY, player.posZ, SoundEvents.ENTITY_PLAYER_ATTACK_SWEEP, player.getSoundCategory(), 1.0F, 1.0F);
+    player.spawnSweepParticles();
+
+    int distance = event.distance;
     boolean hit = false;
-    AxisAlignedBB box = new AxisAlignedBB(entity.posX, entity.posY, entity.posZ, entity.posX + 1.0D, entity.posY + 1.0D, entity.posZ + 1.0D).expand(1.0D, 1.0D, 1.0D);
-    @SuppressWarnings("unchecked")
-    List<Entity> entities = player.worldObj.getEntitiesWithinAABBExcludingEntity(player, box);
+    // we cache the cooldown here since it resets as soon as the first entity is hit
+    for(Entity entity : getAoeEntities(player, target, event)) {
+      if(distance < 0 || entity.getDistanceToEntity(target) <= distance) {
+        hit |= ToolHelper.attackEntity(stack, this, player, entity, null, true);
+      }
+    }
 
-    for(Entity target : entities) {
-      hit |= ToolHelper.attackEntity(stack, this, player, target);
+    if(hit) {
+      player.resetCooldown();
     }
 
+    // subtract the default box and then half as this number is the amount to increase the box by
     return hit;
   }
 
+  private List<Entity> getAoeEntities(EntityPlayer player, Entity target, TinkerToolEvent.ExtraBlockBreak event) {
+    int width = (event.width - 1) / 2;
+    int height = (event.width - 1) / 2;
+    AxisAlignedBB box = new AxisAlignedBB(target.posX, target.posY, target.posZ, target.posX + 1.0D, target.posY + 1.0D, target.posZ + 1.0D).expand(width, height, width);
+
+    return player.worldObj.getEntitiesWithinAABBExcludingEntity(player, box);
+  }
+
   @Nonnull
   @Override
-  public ActionResult<ItemStack> onItemRightClick(@Nonnull ItemStack itemStack, World world, EntityPlayer player, EnumHand hand) {
-    BlockPos start = new BlockPos(player);
-
-    int d = 3;
-    int distance = 3;
-
-    for(int xp = start.getX() - d; xp != start.getX() + d; xp++) {
-      for(int yp = start.getY() - d; yp != start.getY() + d; yp++) {
-        for(int zp = start.getZ() - d; zp != start.getZ() + d; zp++) {
-          if(MathHelper.abs_int(xp - start.getX()) + MathHelper.abs_int(yp - start.getY()) + MathHelper
-              .abs_int(zp - start.getZ()) >= distance) {
-            continue;
+  public ActionResult<ItemStack> onItemRightClick(@Nonnull ItemStack stack, World world, EntityPlayer player, EnumHand hand) {
+    if(ToolHelper.isBroken(stack)) {
+      return ActionResult.newResult(EnumActionResult.FAIL, stack);
+    }
+
+    RayTraceResult trace = this.rayTrace(world, player, false);
```

Comment:
This needs to pass true as the last parameter so it stops at crops (non-opaque-solid blocks)